scale/recipe/models.py

```diff
@@ -212,81 +287,64 @@ class Recipe(models.Model):
 
     class Meta(object):
-        '''meta information for the db'''
+        """meta information for the db"""
         db_table = 'recipe'
 
 
 class RecipeJobManager(models.Manager):
-    '''Provides additional methods for handling jobs linked to a recipe
-    '''
+    """Provides additional methods for handling jobs linked to a recipe
+    """
 
-    @transaction.atomic
-    def get_recipe_jobs(self, recipe_id, jobs_related=False, jobs_lock=False):
-        '''Returns the recipe_job models for the given recipe ID. Each recipe_job model with have its related job model
-        populated.
+    def get_recipe_data(self, recipe_ids):
+        """Returns the recipe, recipe_job, and job models for the given recipe IDs with the recipe/job type and revision
+        models included
 
-        :param recipe_id: The recipe ID
-        :type recipe_id: int
-        :param jobs_related: Whether to include the related models (job_type, job_type_rev) on each related job model
-        :type jobs_related: bool
-        :param jobs_lock: Whether to obtain a select_for_update() lock on each related job model
-        :type jobs_lock: bool
-        :returns: The list of recipe jobs
-        :rtype: list of :class:`recipe.models.RecipeJob`
-        '''
-
-        recipe_job_query = RecipeJob.objects.select_related('job').filter(recipe_id=recipe_id)
-        recipe_jobs = list(recipe_job_query.iterator())
-
-        if jobs_related or jobs_lock:
-            job_ids = []
-            jobs = {}
-            for recipe_job in recipe_jobs:
-                job_ids.append(recipe_job.job_id)
-            # Query job models
-            job_qry = Job.objects.all()
-            if jobs_lock and jobs_related:
-                # Grab locks here and do select_related in separate query
-                Job.objects.select_for_update().filter(id__in=job_ids).order_by('id')
-                job_qry = job_qry.select_related('job_type', 'job_type_rev')
-            elif jobs_related:
-                job_qry = job_qry.select_related('job_type', 'job_type_rev')
-            elif jobs_lock:
-                job_qry = job_qry.select_for_update().order_by('id')
-            for job in job_qry.filter(id__in=job_ids):
-                jobs[job.id] = job
-            # Re-populate the job fields with the updated job models
-            for recipe_job in recipe_jobs:
-                recipe_job.job = jobs[recipe_job.job_id]
-
-        return recipe_jobs
+        :param recipe_ids: The recipe IDs
+        :type recipe_ids: [int]
+        :returns: Dict where each recipe ID maps to a tuple of its recipe model and a list of its recipe_job models
+        :rtype: {int: (:class:`recipe.models.Recipe`, [:class:`recipe.models.RecipeJob`])}
+        """
+
+        # Call get_my_handlers() and organize them, lock dependent jobs, and call get_my_handlers() and organize them
+        recipes = {}  # {Recipe ID: (Recipe, [Recipe job])}
```

Comment:
This return value based on tuples is a bit confusing to follow with all the 0 and 1 based array references in the caller. I think it would be better to use a named tuple for this type of use case: https://docs.python.org/2/library/collections.html#namedtuple-factory-function-for-tuples-with-named-fields